common/presenters/framework-flags-to-tag-list.js

```diff
@@ -0,0 +1,20 @@
+const { sortBy, uniqBy, kebabCase } = require('lodash')
+
+const { FRAMEWORKS } = require('../../config')
+
+function frameworkFlagsToTagList(flags = [], hrefPrefix = '') {
+  const tags = uniqBy(flags, 'title').map(({ flag_type: category, title }) => {
+    const settings = FRAMEWORKS.FLAG_SETTINGS[category] || {}
```

Comment:
_DISCLAIMER: opinion-based comment_.

The function is not pure and it's fairly hard to test (as highlighted by the use of `proxyrequire`).

The function can be easily refactored to be pure by passing the settings as an extra argument or as a high order function.

Example:

```js
function createFrameworkFlagsToTagList(settings = {}) {
  return (flags = [], hrefPrefix = "") => {
    return {
      text: title,
      href: `${hrefPrefix}#${kebabCase(title)}`,
      classes: settings.tagClass || '',
      sortOrder: settings.sortOrder || null,
    }
  }
}
```

Testing the function it's easy (no need for `proxyrequire`) and the same input produces the same output.

The consequences of this change are more complex, though.
You will have to pass down the configuration instead of requiring it locally.

Both options require trade-offs, however, using pure functions means less magic and easier testing.

Reply:
@danielepolencic Yep, I do think that is a much nicer approach. Thanks for just talking through with me too. However, not wanting to derail this change too much as I think the ideal approach would mean refactoring both the presenter and controllers I'm going to write a separate issue if you're happy with that?

Reply:
Added an issue: #696